lib/leche.js fails with latest version of eslint (0.11.0)

lib/leche.js
  166:17  error  Don't make functions within a loop           no-loop-func
  166:26  error  key is already declared in the upper scope   no-shadow
  167:12  error  Don't make functions within a loop           no-loop-func
  220:30  error  Don't make functions within a loop           no-loop-func
  220:39  error  name is already declared in the upper scope  no-shadow
  221:12  error  Don't make functions within a loop           no-loop-func

--- a/lib/leche.js
+++ b/lib/leche.js
@@ -143,7 +143,7 @@ module.exports = {
 		var fake = createObject(template);
 
 		for (var key in fake) {
-			// jshint loopfunc: true
+			/*eslint-disable no-loop-func*/
 
 			// intentionally don't use hasOwnProperty() to get all prototype methods
 			if (isAccessorProperty(template, key)) {	// must check against template, not fake
@@ -163,12 +163,14 @@ module.exports = {
 				});
 
 			} else if (typeof fake[key] === 'function') {
-				fake[key] = (function(key) {
+				fake[key] = (function(methodKey) {
 					return function() {
-						throw new Error('Unexpected call to method "' + key + '".');
+						throw new Error('Unexpected call to method "' + methodKey + '".');
 					};
 				}(key));
 			}
+
+			/*eslint-enable no-loop-func*/
 		}
 
 		return fake;
@@ -215,12 +217,12 @@ module.exports = {
 		 */
 		for (var name in namedDataset) {
 			if (namedDataset.hasOwnProperty(name)) {
-				//jshint loopfunc:true
+				/*eslint-disable no-loop-func*/
 
-				describe('with ' + name, (function(name) {
+				describe('with ' + name, (function(dataName) {
 					return function() {
 
-						var args = namedDataset[name];
+						var args = namedDataset[dataName];
 
 						if (!(args instanceof Array)) {
 							args = [args];
@@ -229,6 +231,8 @@ module.exports = {
 						testFunction.apply(this, args);
 					};
 				}(name)));
+
+				/*eslint-enable no-loop-func*/
 			}
 		}
 	}

--- a/package.json
+++ b/package.json
@@ -30,7 +30,7 @@
   "devDependencies": {
     "browserify": "^5.11.2",
     "chai": "^1.9.1",
-    "eslint": "^0.8.1",
+    "eslint": "^0.11.0",
     "istanbul": "^0.3.2",
     "jsdoc": "^3.3.0-alpha9",
     "jsonlint": "^1.6.2",

